src/constraint_tree.jl

```diff
@@ -191,7 +191,7 @@ interval bound for all variables, it is impossible to use a tuple (since its
 length is 2); in such case use `bound = Ref((minimum, maximum))`, which has the
 correct length.
 """
-function variables(; keys::Vector{Symbol}, bounds = nothing)
+function variables(; keys::Union{Vector{Symbol},Set{Symbol}}, bounds = nothing)
```

Comment:
Do we really want a Set here? (it makes the order of variables in the created ConstraintTree basically unpredictable, which might be a tripwire in case people rely on the numbering, like we do every here and there for simplicity....)

Might be useful to just have AbstractVector here or so